src/main/java/org/folio/circulation/domain/CreateRequestService.java

```diff
@@ -36,11 +44,18 @@ public CreateRequestService(RequestRepository requestRepository, RequestPolicyRe
       .next(RequestServiceUtility::refuseWhenUserHasAlreadyRequestedItem)
       .after(requestLoanValidator::refuseWhenUserHasAlreadyBeenLoanedItem)
       .thenComposeAsync(r -> r.after(requestPolicyRepository::lookupRequestPolicy))
+      .thenComposeAsync(getTimeZoneForRequestRelatedRecords())
       .thenApply(r -> r.next(RequestServiceUtility::refuseWhenRequestCannotBeFulfilled))
       .thenComposeAsync(r -> r.after(updateUponRequest.updateItem::onRequestCreateOrUpdate))
       .thenComposeAsync(r -> r.after(updateUponRequest.updateLoan::onRequestCreateOrUpdate))
       .thenComposeAsync(r -> r.after(requestRepository::create))
       .thenComposeAsync(r -> r.after(updateUponRequest.updateRequestQueue::onCreate))
       .thenApply(r -> r.next(requestNoticeSender::sendNoticeOnRequestCreated));
   }
+
+  private Function<Result<RequestAndRelatedRecords>, CompletionStage<Result<RequestAndRelatedRecords>>> getTimeZoneForRequestRelatedRecords() {
+    return r -> r.combineAfter(
+      unused -> configurationRepository.findTimeZoneConfiguration(),
```

Comment:
We could come up with a convention, typically folks use meaningless single letter variables.

I'd prefer to remove the issue entirely. However I didn't have chance to investigate an alternative. It might be that it is possible to add an overload that takes a supplier instead of a function. Other approaches I've thought of are likely to have type erasure challenges in Java